NetSDK/Services/Parsing/Classes/CompareMatcher.cs

```diff
@@ -3,29 +3,33 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
+using Splitio.CommonLibraries;
 
 namespace Splitio.Services.Parsing
 {
-    public abstract class CompareMatcher: IMatcher
+    public abstract class CompareMatcher:IMatcher
     {
         protected DataTypeEnum? dataType;
         protected long value;
         protected long start;
         protected long end;
-
         public bool Match(string key)
         {
             switch (dataType)
             {
                 case DataTypeEnum.DATETIME:
-                    return MatchDate(key);
+                    var date = key.ToDateTime();
+                    return date != null ? Match(date.Value) : false;
                 case DataTypeEnum.NUMBER:
-                    return MatchNumber(key);
+                    long number;
+                    var result = long.TryParse(key, out number);
```

Comment:
Covered in unit tests : https://github.com/splitio/.net-client/blob/65aa2a6c70e045bd59e36cd9c1022176c2814a8f/Net%20SDK%20Unit%20Tests/Unit%20Tests/Matchers/EqualToMatcherTests.cs#L34